includes/gateway/class-omise-payment-billpayment-tesco.php

```diff
@@ -109,32 +110,109 @@ public function result( $order_id, $order, $charge ) {
+					if ( 'email' === $context ) {
+						$barcode_html = $this->barcode_svg_to_html( $barcode_svg );
+						echo $barcode_html;
+					} else {
+						echo $barcode_svg;
+					}
+					?>
 				</div>
-				<small class="omise-billpayment-tesco-reference-number">|&nbsp;&nbsp;&nbsp; <?php echo $tax_id; ?> &nbsp;&nbsp;&nbsp; 00 &nbsp;&nbsp;&nbsp; <?php echo $ref_1; ?> &nbsp;&nbsp;&nbsp; <?php echo $ref_2; ?> &nbsp;&nbsp;&nbsp; <?php echo $amount; ?></small>
+				<small class="omise-billpayment-tesco-reference-number">
+					<?php
+					echo sprintf(
+						'| &nbsp; %1$s &nbsp; 00 &nbsp; %2$s &nbsp; %3$s &nbsp; %4$s',
+						$charge['source']['references']['omise_tax_id'],
+						$charge['source']['references']['reference_number_1'],
+						$charge['source']['references']['reference_number_2'],
+						$charge['amount']
+					);
+					?>
+				</small>
 			</div>
 			<?php
 		}
+
+		/**
+		 * @param WC_Order $order
+		 *
+		 * @see   woocommerce/templates/emails/email-order-details.php
+		 * @see   woocommerce/templates/emails/plain/email-order-details.php
+		 */
+		public function email_barcode( $order ) {
+			$this->display_barcode( $order, 'email' );
+		}
+
+		/**
+		 * Convert a given SVG Bill Payment Tesco's barcode to HTML format.
+		 *
+		 * Note that the SVG barcode contains with the following structure:
+		 *
+		 * <?xml version="1.0" encoding="UTF-8"?>
+		 * <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="515px" height="90px" viewBox="0 0 515 90" version="1.1" preserveAspectRatio="none">
+		 *   <title>** reference number **</title>
+		 *   <g id="canvas">
+		 *     <rect x="0" y="0" width="515px" height="90px" fill="#fff" />
+		 *     <g id="barcode" fill="#000">
+		 *       <rect x="20" y="20" width="2px" height="50px" />
+		 *       ... (repeat <rect> node for displaying barcode) ...
+		 *       <rect x="493" y="20" width="2px" height="50px" />
+		 *     </g>
+		 *   </g>
+		 * </svg>
+		 *
+		 * The following code in this method is to read all <rect> nodes' attributes under the <g id="barcode"></g>
+		 * in order to replicate the barcode in HTML <div></div> element.
+		 *
+		 * @param  string $barcode_svg
+		 *
+		 * @return string  of a generated Bill Payment Tesco's barcode in HTML format.
+	     */
+		public function barcode_svg_to_html( $barcode_svg ) {
+			$xml       = new SimpleXMLElement( $barcode_svg );
+			$xhtml     = new DOMDocument();
+			$prevX     = 0;
+			$prevWidth = 0;
+
+			// Read data from all <rect> nodes.
+			foreach ( $xml->g->g->children() as $rect ) {
+				$attributes = $rect->attributes();
+				$width      = $attributes['width'];
+				$margin     = ( $attributes['x'] - $prevX - $prevWidth ) . 'px';
+
+				// Set HTML attributes based on <rect> node's attributes.
+				$divRect = $xhtml->createElement( 'div' );
+				$divRect->setAttribute( 'style', "float: left; position: relative; height: 50px; border-left: $width solid #000000; width: 0; margin-left: $margin" );
```

Comment:
Do we have a potential problem with barcodes not being visible on themes that have a black background? The HTML here does not seem to have a white background behind the bars

Reply:
@jonrandy if you ask me, super rare. There is a white background at the confirmation email by itself.
Well, I can add a white background here to make sure that it will prevent for those edge cases, probably a new PR after #126 should be ok? (just to avoid merging conflict).

Reply:
New PR is ok, but should probably be done before release

Reply:
sure, right after #125 and #126. I'm planning to submit one more for adjusting UI and add some more missing info.